meson.build

```diff
@@ -679,6 +679,10 @@ if time_epoch == -1
         source_date_epoch = run_command('sh', ['-c', 'echo "$SOURCE_DATE_EPOCH"']).stdout().strip()
         if source_date_epoch != ''
                 time_epoch = source_date_epoch.to_int()
+        elif run_command('test', '-d', '.git').returncode() == 0
```

Comment:
This looks almost OK, but will break when cwd is a git worktree. Please change that to `test -e`.

Reply:
Or maybe just run the first `git` command and go to the fallback if stdout is empty? This will also help if `git` is not available (e.g. building an existing checkout in a container).

Reply:
Right, the case where `git` is not available needs to work correctly too.
But, we can't just execute `git` without checking for `.git`, because we might be in a git-less checkout in some higher level directory and git recurses up.

Reply:
I think what I have now addresses both points.